subprojects/base-services/src/main/java/org/gradle/api/JavaVersion.java

```diff
@@ -54,10 +54,10 @@ public static JavaVersion toVersion(Object value) throws IllegalArgumentExceptio
         }
 
         String name = value.toString();
-        Matcher matcher = Pattern.compile("(\\d{1,2})(-.+)?").matcher(name);
+        Matcher matcher = Pattern.compile("(\\d{1,2})([-\\D].+)?").matcher(name);
```

Comment:
`\\D` already covers `-`, no need to add both. So instead of `[-\\D]` a plain `\\D` should be enough.

Reply:
Good catch!